src/gp/src/GPMSA.C

```diff
@@ -899,13 +899,17 @@ GPMSAFactory<V, M>::setUpEmulator()
   // Observation precision matrix
   M Wy(env, Wy_row_map, Wyrows);
 
+  m_observationErrorMatrix.reset(new M(env, Wy_row_map, Wyrows));
+
   for (unsigned int ex = 0; ex != m_numExperiments; ++ex)
     {
       const M & D_i = m_discrepancyMatrices[ex];
 
+      const M & Sigma_i = *m_observationErrorMatrices[ex];
```

Comment:
You've got the handy readably-named reference here, why not use it when taking the inverse below?